src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -1996,8 +1996,8 @@ private bool HasDescendantsWith(XamlObjectDefinition xamlObjectDefinition, Func<
 				// the fully qualified type name.
 				// In this case, we go through this code path as it's much more efficient than FindType.
 				var baseTypeString = $"{ns}.{xamlType.Name}";
-				findType = _metadataHelper.FindTypeByFullName(baseTypeString) as INamedTypeSymbol;
-				findType ??= _metadataHelper.FindTypeByFullName(baseTypeString + "Extension") as INamedTypeSymbol; // Support shortened syntax
+				findType = _metadataHelper.FindTypeByFullName(baseTypeString + "Extension") as INamedTypeSymbol; // Support shortened syntax
+				findType ??= _metadataHelper.FindTypeByFullName(baseTypeString) as INamedTypeSymbol;
```

Comment:
this is indeed the correct lookup order according to microsoft:
> The order of lookup is to look for the Extension-suffixed class name first and then look for the class name without the Extension suffix.
> -- https://learn.microsoft.com/en-us/dotnet/desktop/xaml-services/markup-extensions-overview#defining-the-support-type-for-a-custom-markup-extension

that said, the problem isn't just in the order, but also in not type checking immediately before locking in the result:
```suggestion
				// Support shortened syntax
				findType = _metadataHelper.FindTypeByFullName(baseTypeString + "Extension") is INamedTypeSymbol type2 && type2.Is(Generation.MarkupExtensionSymbol.Value) ? type2 : null;
				findType ??= _metadataHelper.FindTypeByFullName(baseTypeString) is INamedTypeSymbol type1 && type1.Is(Generation.MarkupExtensionSymbol.Value) ? type1 : null;
```

Reply:
the entire method could also use some refactoring too to streamline it better, and to avoid the useless last `Is`-check